main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -210,24 +210,37 @@ object TypeReduction {
    * a ConstructorNotFound object. The working process is similar to retrieveMethod and differs in the selection of candidate constructors.
    */
   private def retrieveConstructor(clazz: Class[_], ts: List[Type], loc: SourceLocation): JavaConstructorResolutionResult = {
-    val candidateConstructors = clazz.getConstructors.filter(c => isCandidateConstructor(c, ts))
+    val candidateConstructors = clazz.getConstructors.filter(c => isCandidateConstructor(c, ts)).toList
 
-    candidateConstructors.length match {
-      case 0 => JavaConstructorResolutionResult.ConstructorNotFound
-      case 1 =>
-        val tpe = Type.Cst(TypeConstructor.JvmConstructor(candidateConstructors.head), loc)
+    candidateConstructors match {
+      // Case 1: No such constructor. Error.
+      case Nil => JavaConstructorResolutionResult.ConstructorNotFound
```

Comment:
It adds linear a linear pass that wasn't there before in the cases for 0 and 1. Is this okay? i guess retrieveConstructor is called rarely?

Reply:
It's OK IMO since we're passing linearly over the constructors already in the `filter` function